citadel_types/src/crypto/mod.rs

```diff
@@ -67,6 +67,16 @@ pub enum SecrecyMode {
     BestEffort,
 }
 
+impl From<u8> for SecrecyMode {
+    fn from(value: u8) -> Self {
+        match value {
+            0 => Self::Perfect,
+            1 => Self::BestEffort,
+            _ => panic!("Cannot cast `{}` into SecrecyMode", value),
+        }
+    }
+}
+
```

Comment:
The rustic way to handle cases where there is a possibility `From` or `Into` fails is to instead implement the trait `TryFrom` or `TryInto`. Though if `T: TryFrom<U>` then automatically `U: TryInto<T>`; in other words, if `T` can potentially be derived from `U`, then `U` can potentially be turned into `T`. So, implement `From` (for no potential fallibility) or `TryFrom` (w/ potential fallibility) to get the free automatic implementatic of `Into` and/or `TryInto`